src/coordination/coordinator_instance.cpp

```diff
@@ -670,6 +725,61 @@ auto CoordinatorInstance::RegisterReplicationInstance(CoordinatorToReplicaConfig
   return RegisterInstanceCoordinatorStatus::SUCCESS;
 }
 
+auto CoordinatorInstance::DemoteInstanceToReplica(std::string_view instance_name) -> DemoteInstanceCoordinatorStatus {
+  auto lock = std::lock_guard{coord_instance_lock_};
+
+  if (!is_leader_ready_) {
```

Comment:
When the leader isn't ready?

Reply:
If it is just becoming leader, RAFT server elected it

Reply:
write that down